arch/x86/core/pcie.c

```diff
@@ -236,6 +236,74 @@ static inline void _write_pcie_irq_data(pcie_bdf_t bdf, uint32_t data)
 	pcie_conf_write(bdf, PCIE_CONF_INTR, data);
 }
 
+#if !defined(CONFIG_INTEL_VTD_ICTL)
+static unsigned int get_msi_aligned_vector(unsigned int priority,
+						msi_vector_t *vectors,
+						uint8_t n_vector)
+{
+	const int MAX_PRIORITY = 13;
+	int vector = 0, prev_vector = -1, start_vector;
+	int i;
+
+	while (vector != -1) {
+		vector = z_x86_allocate_vector(priority, prev_vector);
+		if ((vector == -1) && (priority <= MAX_PRIORITY)) {
+#if defined (CONFIG_MULTI_MSI_ANY_PRIORITY)
```

Comment:
to declutter things when there is a lot of ifdefs everywhere. Use if (IS_ENABLED()) {} else {} whenever possible (when no function/variable are out of the definition)